src/main/kotlin/io/github/inductiveautomation/kindling/log/LogPanel.kt

```diff
@@ -87,7 +88,10 @@ class LogPanel(
     private val tableScrollPane = FlatScrollPane(table)
 
     private val sidebar = FilterSidebar(
-        NamePanel(rawData),
+        if (rawData.first() is SystemLogEvent && ShowLogTree.currentValue) {
+            @Suppress("UNCHECKED_CAST")
+            NameTreePanel(rawData as List<SystemLogEvent>)
+        } else NamePanel(rawData),
```

Comment:
You could introduce a 'wrapper' component, that itself pays attention to the ShowLogTree preference.
Add it to the filter sidebar here, and then internally have it 'swap' between the two panel approaches.

Reply:
This does get a fair bit more complicated, especially to Josh's point of supporting hot-swap between filters...but I would also argue it's in line with user expectations. It should be possible to come up with an abstract class that acts a model that's suitable for both display components, theoretically.